pandas/core/frame.py

```diff
@@ -3920,17 +3920,7 @@ def _combine_frame(self, other, func, fill_value=None, level=None):
         new_index, new_columns = this.index, this.columns
 
         def _arith_op(left, right):
```

Comment:
you might be able to simplify this even more here (IOW remove _arith_op) and just in-line it

Reply:
This overlaps with #19611.  This will probably get in before that does, so I'll take a look at this suggestion after rebasing there.

Reply:
aok